app/Kconfig

```
@@ -316,6 +316,14 @@ config ZMK_RGB_UNDERGLOW_ON_START
 	bool "RGB underglow starts on by default"
 	default y
 
+config ZMK_RGB_UNDERGLOW_AUTO_OFF_IDLE
+	bool "Turn off RGB underglow when keyboard goes into idle state"
+	default y
+
+config ZMK_RGB_UNDERGLOW_AUTO_OFF_USB
+	bool "Turn off RGB underglow when USB is disconnected"
+	default n
```

This should skip the redundant `default n` (see https://docs.zephyrproject.org/2.5.0/guides/kconfig/tips.html#redundant-defaults) and include a depends on USB, e.g.:

```suggestion
config ZMK_RGB_UNDERGLOW_AUTO_OFF_USB
	bool "Turn off RGB underglow when USB is disconnected"
	depends on USB
```

---

We cannot use a single variable. Consider the following situation when both `AUTO_OFF_IDLE` and `AUTO_OFF_USB` are enabled.

* initially `state=on`
* unplug USB: `state=off`, `prev_state=on`
* idle state: nothing happens
* back from idle state
  expected behaviour: nothing happens
  actual behaviour: RGB turns on since `prev_state=on`